src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(
+    name,
+    health,
+  ) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(
+    name,
+    health,
+    hidden,
+  ) {
+    super(
+      name,
+      health,
+      hidden,);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
```

Comment:
```suggestion
    this.hidden = true;
```
the method with the name "hide" should only hide